### PR TITLE
[CFM-1332] Configure project permissions.

### PR DESCRIPTION
**Pull request by anvmn**
https://issuetracker.amplexor.com/jira/browse/CFM-1332

### CODE REVIEW

File: project/profiles/capacity4more/modules/c4m/features_og/c4m_features_og_members/c4m_features_og_members.module [hunk 1]
```diff
@@ -516,12 +516,21 @@ function _c4m_features_og_members_get_user_type($group = NULL, $user = NULL) {
     return C4M_USER_TYPE_GO;
   }
 
-  // The user has the group role 'administrator member'.
-  if (in_array('administrator member', og_get_user_roles('node', $group->nid, $user->uid))) {
-    return C4M_USER_TYPE_GA;
-  }
+  if ($group->type == 'group') {
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/features_og/c4m_features_og_members/c4m_features_og_members.module]: the condition should only check if it's 'project', no need the `else` afterward ...

File: project/profiles/capacity4more/modules/c4m/og/c4m_og/c4m_og.module [hunk 2]
```diff
@@ -2454,7 +2455,8 @@ function c4m_og_create_node_access($node, $account) {
   // When creating, $node is string that indicates bundle name.
   if (og_is_group_type('node', $node)) {
 
-    return user_is_anonymous() ? NODE_ACCESS_DENY : NODE_ACCESS_ALLOW;
+    // Any authenticated user may create groups and projects.
+    return $account->uid == 0 ? NODE_ACCESS_DENY : NODE_ACCESS_ALLOW;
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/og/c4m_og/c4m_og.module]: `===`

**Comment by ordavidil** [comment]
@anvmn 
Can you please write down what has been changed ?  like, what was not working before that is working now ?
thanks

**Comment by anvmn** [comment]
@ordavidil 
User type resolving extended for projects.
Project permissions (realms and node access) aligned.
Search API filter applied on /projects view.
**Fixing:**
'My projects' top bar presents only the projects user can view.
/projects view presents only the projects user can view.

File: project/profiles/capacity4more/behat/features/bootstrap/FeatureContext/Project.php [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  public function theProjectStatusIsChangedByAdminTo($group_title, $status) {
+    $steps = array();
+
+    $group = $this->loadGroupByTitleAndType($group_title, 'project');
+    $steps[] = new Step\When('I am logged in as user "admin"');
+    $steps[] = new Step\When('I visit "/node/' . $group->nid . '/edit"');
+    $steps[] = new Step\When('I select "' . $status . '" from "edit-c4m-og-status-und"');
+    $steps[] = new Step\When('I press "Save"');
+    $steps[] = new Step\When('I should see "Project ' . $group_title . ' has been updated."');
+    return $steps;
```
Comment (ordavidil) [inline project/profiles/capacity4more/behat/features/bootstrap/FeatureContext/Project.php]: add line spacing above

**Comment by anvmn** [comment]
Closing in favor of https://github.com/capacity4dev/capacity4more/pull/1612

**Pull request closed without merging**